core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala

```diff
@@ -48,83 +52,94 @@ private[spark] trait CachedReceiver {
  * @param nAndP the Event Hub name and partition that the receiver is connected to.
  */
 private[client] class CachedEventHubsReceiver private (ehConf: EventHubsConf,
-                                                       nAndP: NameAndPartition)
+                                                       nAndP: NameAndPartition,
+                                                       startSeqNo: SequenceNumber)
     extends Logging {
 
   import org.apache.spark.eventhubs._
 
   private lazy val client: EventHubClient = ClientConnectionPool.borrowClient(ehConf)
 
+  private var _receiver: Future[PartitionReceiver] = _
+  private def receiver: Future[PartitionReceiver] = {
+    if (_receiver == null) {
```

Comment:
When could the receiver be null? It is created right at the initialization of the class. I'm not super familiar with the execution path of Spark, but does it re-use closed receivers? This would be an anti pattern.

Reply:
Maybe we can do something smart and create a wrapper around it?

Reply:
**The problem:** Let's say we have our receiver for partition `x` cached on executor `y`. We consume up to sequence number 2000. Then an executor is added, so now tasks for partition `x` are assigned to executor `z`. There's a cache miss and a new cache is made. Let's say we consume until sequence number 3000. Then, just after that, the newly added executor is removed by Spark. Now, partition `x` tasks are back to executor `y`. There's a cache hit. We expect to consume sequence number 3001, but the next queued event is 2001 - that's where we left off in this cache. 

My current solution is to close the receiver and create a new one at the right spot. Though, after thinking about this, a better solution may be to just consume events until we get to the expected sequence number.